src/saml.ts

```diff
@@ -1226,13 +1226,23 @@ class SAML {
         if (!restriction.Audience || !restriction.Audience[0] || !restriction.Audience[0]._) {
           return new Error("SAML assertion AudienceRestriction has no Audience value");
         }
-        if (restriction.Audience[0]._ !== expectedAudience) {
-          return new Error(
-            "SAML assertion audience mismatch. Expected: " +
-              expectedAudience +
-              " Received: " +
-              restriction.Audience[0]._,
-          );
+        const receivedAudiences = [];
+        let i = 0;
+        while (i < restriction.Audience.length) {
```

Comment:
Why can't you use `.some()`?

Reply:
@cjbarth I didn't know it could do that. Rewrote using `.some()`.